src/components/SurveyAnswer/TextfieldAnswer.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import { SurveyAnswerProps } from '.';
+
+const TextfieldAnswer = ({
+  question,
+  currentResponse,
+  setResponse,
+}: SurveyAnswerProps) => {
+  const setAnswerValue = (answerId: string) => {
+    const answerValue = currentResponse?.answers.find(
+      (answer) => answer.id === answerId
+    )?.answer;
+
+    return answerValue || '';
+  };
```

Comment:
From the name of this function, I though it was changing the "state variable `answerValue`". Let's keep `SetXXX` naming for the `UseState` setters 😇 

```suggestion
  const answerValue = (answerId: string) => {
    const currentValue = currentResponse?.answers.find(
      (answer) => answer.id === answerId
    )?.answer;

    return currentValue || '';
  };
```